packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -118,7 +109,10 @@ export class H5WasmApi extends ProviderApi {
     const file = await this.file;
 
     const h5wEntity = file.get(path);
-    assertNonNull(h5wEntity, `No entity found at ${path}`);
+    assertNonNull(
+      h5wEntity,
+      path === '/' ? `Expected valid HDF5 file` : `No entity found at ${path}`
```

Comment:
If I drop a JSON file, for instance, `file.get('/')` logs a warning and returns `null`, so I'm generalising here that when we get `null` for the root entity, it's because h5wasm had a problem reading the file (most likely because it's not an HDF5 file).